### PR TITLE
[CP-4659] feat: drop direct dependency on numpy and pandas

### PR DESCRIPTION
**Pull request by rplopes**
Part of https://datacamp.atlassian.net/browse/CP-4659

This makes numpy and pandas no longer required dependencies, meaning that they won't be automatically installed when installing pythonwhat.

The only usage of any of them on anything other than comparing numpy/pandas objects (i.e. expected usage without the need to have those packages already imported anyway) is a `np.testing.assert_equal(x, y)` to compare `dict`, `list` and `tuple` objects, and a `np.array_equal(list(x), list(y))` to compare `map` and `filter` objects. Those calls got refactored to not need numpy to achieve the same behavior.

With this change, users who relied on the instance of numpy and/or pandas installed by pythonwhat now need to install it themselves. This shouldn't affect most users, as it's good practice to always install one's own dependencies instead of having to rely on having them installed by other dependencies.

### CODE REVIEW

File: pythonwhat/Test.py [hunk 1]
```diff
@@ -117,34 +123,64 @@ def is_equal(x, y):
...
+
         # Delay importing pandas / numpy until absolutely necessary. This is important for performance in Pyodide.
-        import pandas as pd
-        from pandas.testing import assert_frame_equal, assert_series_equal
-        import numpy as np
-
-        if areinstance(x, y, (np.ndarray, dict, list, tuple)):
-            np.testing.assert_equal(x, y)
-            return True
-        elif areinstance(x, y, (map, filter)):
```
Comment (RuiGuedes) [inline pythonwhat/Test.py:134]: Are we still covering this conditional statement?

File: pythonwhat/Test.py
Comment (RuiGuedes) [inline pythonwhat/Test.py:1]: How confident are you that the refactor here leads to the same behavior?

File: pythonwhat/Test.py (hunk 1 above)
Comment (rplopes) [reply pythonwhat/Test.py:134]: Yes, L139

File: pythonwhat/Test.py
Comment (rplopes) [reply pythonwhat/Test.py:1]: The tests pass and the few manual tests I did in Intro to Python seemed fine, though to be honest I may not have tested enough weird edge cases where there's a greater chance of there being something wrong. I can always look out for more SCTs and check if they still return the expected result. Fortunately, the worst case scenario is if there's a change in behavior for some edge case in some course, then the course build will fail when attempting to upgrade its course image to one with this updated pythonwhat version.

Comment (rplopes) [reply pythonwhat/Test.py:1]: I've tested all SCTs for Intro to Python and Intermediate Python, and there was no regression, so that raises the confidence for now. Once we roll this out to more courses, we can always react at that time to any failed validation.

Comment (RuiGuedes) [reply pythonwhat/Test.py:1]: Thanks for double-checking and for the info.

**Review by RuiGuedes — APPROVED** [review]
